config/config.go

```diff
@@ -36,6 +40,8 @@ type (
 )
 
 // IsDataDogConfigValid method to verify if configuration values are correct
+//
+// Deprecated: Use Validate()
```

Comment:
When deprecating it, we should refactor this function to use the new Validate function internally

Reply:
Or we can document the code as frozen, that is probably even better.

Do we want `Validate()` to be compatible with `IsDataDogConfigValid()` until v3?

Reply:
We can omit the deprecation until v3, but use the new `Validate` internally?

Reply:
I think we should deprecate, but we can use it internally

Reply:
Call validate internally now